sources/packages/backend/apps/queue-consumers/src/processors/schedulers/sfas-integration/_tests_/sims-to-sfas-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,249 @@
+    // and not having to update the database for each test case
+    // which could potentially have various columns to be updated.
+    simsDataUpdatedDate = addMilliSeconds(10, latestBridgeFileDate);
+    mockedCurrentDate = addMilliSeconds(10, simsDataUpdatedDate);
+  });
+
+  afterEach(async () => {
+    // Reset the current date mock.
+    MockDate.reset();
+  });
+
+  it(
+    "Should generate a SIMS to SFAS bridge file when there is an update on student data" +
+      " between the most recent bridge file date and the current bridge file execution date.",
+    async () => {
+      // Arrange
+      // Create bridge file log.
+      await db.sfasBridgeLog.insert({
+        referenceDate: latestBridgeFileDate,
+        generatedFileName: "DummyFileName.TXT",
+      });
+
+      // Student created with expected first name, last name and more importantly the updated date
+      // to fall between the most recent bridge file date and the mocked current date.
+      const student = await createStudentWithExpectedData(simsDataUpdatedDate);
+      // Student has submitted an application.
+      await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { student },
+        { offeringIntensity: OfferingIntensity.partTime },
+      );
+
+      // Queued job.
+      const mockedJob = mockBullJob<void>();
+
+      // Mock the current date.
+      MockDate.set(mockedCurrentDate);
+
+      // Expected file name.
+      const expectedFileName = buildExpectedFileName(mockedCurrentDate);
+
+      // Act
+      const processingResult = await processor.generateSFASBridgeFile(
+        mockedJob.job,
+      );
+
+      // Assert
+      // Assert process result.
+      expect(processingResult).toEqual([
+        "Process finalized with success.",
+        "Student records sent: 1.",
+        `Uploaded file name: ${expectedFileName}.`,
+      ]);
+      expect(
+        mockedJob.containLogMessages([
+          `Processing data since ${latestBridgeFileDate} until ${mockedCurrentDate}.`,
+          "Found 1 student(s) with updates.",
+          `SIMS to SFAS file ${expectedFileName} has been uploaded successfully.`,
+          `SIMS to SFAS file log has been created with file name ${expectedFileName} and reference date ${mockedCurrentDate}.`,
+        ]),
+      ).toBe(true);
+      const uploadedFile = getUploadedFile(sftpClientMock);
+      const [header, studentRecord, footer] = uploadedFile.fileLines;
+      expect(header).toBe(buildHeader(mockedCurrentDate));
+      expect(footer).toBe("999000000001");
+      expect(studentRecord).toBe(buildStudentRecord(student));
+      // Check the database for creation of SFAS bridge log.
+      const uploadedFileLog = await db.sfasBridgeLog.findOneBy({
+        generatedFileName: expectedFileName,
+        referenceDate: mockedCurrentDate,
+      });
+      expect(uploadedFileLog.id).toBeGreaterThan(0);
+    },
+  );
+
+  it(
```

Comment:
Can we be consistent about where the white space goes? This description was added twice. I would recommend adding to the end of the lines only.